instrumentation-api-semconv/src/test/java/io/opentelemetry/instrumentation/api/instrumenter/net/PeerServiceAttributesExtractorTest.java

```diff
@@ -94,4 +96,57 @@ void shouldSetPeerNameIfItMatches() {
     assertThat(endAttributes.build())
         .containsOnly(entry(SemanticAttributes.PEER_SERVICE, "myService"));
   }
+  
+  @Test
+  void shouldSetSockPeerNameIfItMatchesAndNoPeerNameProvided() {
+    // given
+    Map<String, String> peerServiceMapping = new HashMap<>();
+    peerServiceMapping.put("example.com", "myService");
+    peerServiceMapping.put("1.2.3.4", "someOtherService");
+
+    PeerServiceAttributesExtractor<String, String> underTest =
+        new PeerServiceAttributesExtractor<>(netAttributesExtractor, peerServiceMapping);
+
+    when(netAttributesExtractor.sockPeerName(any(), any())).thenReturn("example.com");
+
+    Context context = Context.root();
+
+    // when
+    AttributesBuilder startAttributes = Attributes.builder();
+    underTest.onStart(startAttributes, context, "request");
+    AttributesBuilder endAttributes = Attributes.builder();
+    underTest.onEnd(endAttributes, context, "request", "response", null);
+
+    // then
+    assertThat(startAttributes.build()).isEmpty();
+    assertThat(endAttributes.build())
+        .containsOnly(entry(SemanticAttributes.PEER_SERVICE, "myService"));
+  }
+
+  @Test
+  void shouldSetNoSockPeerNameIfPeerNameMatches() {
```

Comment:
This test is almost the same as `shouldSetPeerNameIfItMatches`; instead of duplicating it, can you add the final `verify()` to that test case?